### DISCUSSION TITLE
Update and Set Up New Course Errors

### DISCUSSION THREAD

**Discussion by RachelBrodsky** [inducer/relate#874] [General]
Hello, 

My team deployed Relate to an EC2 instance. We were able to upload a course, and we've been able to successfully update the github repo and fetch/update the course. Suddenly, we started seeing this error when we try to update the course (or set up a new course with the same github repo). This is happening on both the EC2 instance and my localhost. 

I deleted all my newly added code and tried again, but I'm still getting this error (even though I was able to successfully update the course yesterday). Do you have any idea how to fix this issue?

<img width="1788" alt="Screen Shot 2022-01-11 at 12 40 06 PM" src="https://user-images.githubusercontent.com/67277602/148993636-e088010f-51de-44ad-b1bd-82e5972458ae.png">
<img width="1577" alt="Screen Shot 2022-01-11 at 12 40 26 PM" src="https://user-images.githubusercontent.com/67277602/148993687-c826727c-b0e5-48f8-95ad-f32821ed8996.png">

![image](https://user-images.githubusercontent.com/67277602/148994159-1f399aa4-f6f6-4ea0-9538-fc54127830d2.png)

**Comment by RachelBrodsky** [comment]
It seems like this is a larger github issue -- uploading the sample course no longer works either. Any ideas on how to address this?

![image](https://user-images.githubusercontent.com/67277602/149000137-302832c4-82a8-4e94-8733-20366054cf95.png)

**Comment by inducer** [comment]
There are two separate issues here.

* https://github.com/inducer/relate/pull/875 should address the first one (Github no longer takes RSA keys with SHA-1).
* https://github.com/inducer/relate/pull/876 should address the other.

**Comment by RachelBrodsky** [comment]
I made those changes and I get this error. Do you know what is causing that?

<img width="782" alt="Screen Shot 2022-01-11 at 5 31 22 PM" src="https://user-images.githubusercontent.com/67277602/149031867-64614750-d0a2-403f-909f-173128b071c3.png">

**Comment by inducer** [comment]
> I made those changes

Which changes? What precisely did you do?

**Reply by RachelBrodsky** [reply]
(#875) 
1) I replaced my poetry.lock and pyproject.toml files with your updated files. 
2) I ran poetry install and poetry shell in my virtual environment to update the library versions

(#876) 
1) I updated course/models.py by replacing  **"<tt>git://github.com/inducer/relate-sample</tt> "** with
            **"<tt>https://github.com/inducer/relate-sample.git</tt> "**
2) I updated doc/misc.rst by replacing **git clone git://github.com/inducer/relate** with **git clone https://github.com/inducer/relate.git**
3) I updated  tests/base_test_mixins.py by replacing git_source": **"git://github.com/inducer/relate-sample",** with
            **"git_source": "https://github.com/inducer/relate-sample.git",**
4) I ran "python manage.py migrate" and "python manage.py makemigrations" to migrate the changes from models.py

Then I ran the application (using python manage.py runserver) and went to set up a new course. I generated an rsa-sha2 using the following command in my terminal: **ssh-keygen -t rsa-sha2-256 -m PEM**. I used the private key in the course creation form, and I put the public key in my repo (as a deployed key). I used the https url to my course repo to try and set up the new course:https://github.com/RachelBrodsky/CIRCUIT_Training.git (I tried with and without the .git extension). When I submitted the form I got the HTTPUnauthorized error.

**Reply by RachelBrodsky** [reply]
For #875 I replaced my entire poetry.lock and pyproject.toml files with the files currently in this repo. For #876 I replaced individual lines.

**Reply by inducer** [reply]
They're both merged now, so `git pull` would likely do the same.

> https://github.com/RachelBrodsky/CIRCUIT_Training.git

Looks like that's a private repo. You'll need to use the SSH address to authenticate, i.e. likely `git@github.com:RachelBrodsky/CIRCUIT_Training`.

**Reply by RachelBrodsky** [reply]
ah, I see. That worked! Thank you!
